### PR TITLE
Add Subset operator

### PR DESCRIPTION
**Pull request by mikeurbanski1**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Adds a pair of operators: "subset" and "not_subset". These will be useful for things like IP address allowlists (or any allowlist).

### CODE REVIEW

File: checkov/common/checks_infra/solvers/attribute_solvers/subset_attribute_solver.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        super().__init__(resource_types=resource_types,
+                         attribute=attribute, value=SubsetAttributeSolver.to_set(value))
+
+    def _get_operation(self, vertex: Dict[str, Any], attribute: Optional[str]) -> bool:
+        attr_val = SubsetAttributeSolver.to_set(vertex.get(attribute))
+        return attr_val.issubset(self.value)
+
+    @staticmethod
+    def to_set(value: Any) -> Set[Any]:
+        if isinstance(value, Set):
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/subset_attribute_solver.py:21]:
```suggestion
        if isinstance(value, set):
```

File: checkov/common/checks_infra/solvers/attribute_solvers/subset_attribute_solver.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            return value
+        elif isinstance(value, (list, dict)):
+            return set(value)
+        else:
+            s = set()
+            s.add(value)
+            return s
```
Comment (gruebel) [inline checkov/common/checks_infra/solvers/attribute_solvers/subset_attribute_solver.py:28]:
```suggestion
            return {value}
```

File: tests/terraform/graph/resources/arrays/main.tf [hunk 3]
```diff
@@ -0,0 +1,46 @@
+resource "x" "pass1" {
+  arr = [
+    "allowed1"
+  ]
+}
+
+resource "x" "pass2" {
+  arr = [
+  ]
+}
```
Comment (gruebel) [inline tests/terraform/graph/resources/arrays/main.tf:10]: not sure about this one, is no value a subset of a list of values? mathematically it is correct, but does it make sense for the user? @JamesWoolfenden any thoughts?

**Review by gruebel — COMMENTED** [review]
nice, great addition

File: tests/terraform/graph/resources/arrays/main.tf (hunk 3 above)
Comment (mikeurbanski1) [reply tests/terraform/graph/resources/arrays/main.tf:10]: Well, it's either this or return a mathematically incorrect value

**Review by gruebel — APPROVED** [review]
🍻

**Pull request merged**